### ISSUE TITLE
[5.0.5] Not working anymore on pictures, only working for videos

### ISSUE THREAD

**Issue by Rocket3192** [theus/instantgram#86]
Hi, since yesterday or maybe the day before, the script isn't working at all anymore for pictures, it does sort of work for videos though
Instagram must have changed something
The weird thing is that it works for posts on instagram.com (home page), but on a post page instagram.com/XXXXXXX it doesn't work nothing happens when I click on the script

(Version 5.0.5)

Using the latest version of Edge (Chromium)

Console : 
 ```
ImageInPostAndModal() [instantgram] 5.0.5 TypeError: Cannot read properties of null (reading 'querySelector')
    at t.execute (<anonymous>:1:6315)
    at Module.<anonymous> (<anonymous>:1:15263)
    at n (<anonymous>:1:123)
    at <anonymous>:1:915
    at <anonymous>:1:924
    at <anonymous>:1:15481
```

Plus, but it wasn't working before either but I never said it here : It doesn't open story pictures

Thank you

**Comment by Rocket3192** [comment]
It also works when pages are like this : 
![image](https://user-images.githubusercontent.com/123319690/224373010-3c959323-13b2-43ef-ab9e-f6ce129174d2.png)


But not when they're like this (opened in new tab) : 
![image](https://user-images.githubusercontent.com/123319690/224372925-43573c7a-e444-48e4-8a52-f250c2c4a6b9.png)

**Comment by juckono** [comment]
Same on Firefox. Nothing happens when clicking a bookmark on a separate post, but works on homepage.

**Comment by eddbaz3** [comment]
Same issue here in the latest version of Chrome Version 111.0.5563.147 (Official Build) (64-bit)

VM81:1 ImageInPostAndModal() [instantgram] 5.0.5 TypeError: Cannot read properties of null (reading 'querySelector')
    at t.execute (<anonymous>:1:6315)
    at Module.<anonymous> (<anonymous>:1:15263)
    at n (<anonymous>:1:123)
    at <anonymous>:1:915
    at <anonymous>:1:924
    at <anonymous>:1:15481

**Comment by rainmakerock** [comment]
I'm using Firefox. Nothing happens when clicking a bookmark on a separate post

**Comment by jeohist** [comment]
I'm getting the same error in Chrome 113.0.5672.92

**Comment by SceptreData** [comment]
My quick fix:

Search the bookmarklet string for the following text:

`document.querySelectorAll('article%5Brole=%22presentation%22%5D')`

Replace `article` with `div` and you should be good to go. 

 I'm still having the issue where I can't download the 3rd image in a set of 4 or more, but I can only try to work it out from the minified code for so long.

**Comment by Wag669** [comment]
> My quick fix: [...]

Replaced article with div and it seems to be working for now 
Thanx!

**Comment by Floyd77** [comment]
> My quick fix:
> 
> Search the bookmarklet string for the following text:
> 
> `document.querySelectorAll('article%5Brole=%22presentation%22%5D')`
> 
> Replace `article` with `div` and you should be good to go.
> 
> I'm still having the issue where I can't download the 3rd image in a set of 4 or more, but I can only try to work it out from the minified code for so long.

Not working on v5.0.6 & Firefox 121 

Console output:
`ImageInPostAndModal() [instantgram] 5.0.6 TypeError: a is null`

It only works with videos and with pages like the first example from @Rocket3192

**Comment by SceptreData** [comment]
Yeah, looks like the gang at meta pushed an update. Markup is a bit different. I wrote a hacky fix last night but it was failing in some unexpected places.
